pkg/band/band.go

```diff
@@ -300,248 +304,368 @@ var usAuBeaconFrequencies = func() (freqs [8]uint64) {
 	return freqs
 }()
 
+func parseChMask(offset uint8, mask ...bool) map[uint8]bool {
+	if len(mask)-1 > int(math.MaxUint8-offset) {
+		panic(fmt.Sprintf("channel mask overflows uint8, offset: %d, mask length: %d", offset, len(mask)))
+	}
+	m := make(map[uint8]bool, len(mask))
+	for i, v := range mask {
+		m[offset+uint8(i)] = v
+	}
+	return m
+}
+
 func parseChMask16(mask [16]bool, cntl uint8) (map[uint8]bool, error) {
-	chans := make(map[uint8]bool, 16)
 	switch cntl {
 	case 0:
-		for i := uint8(0); i < 16; i++ {
-			chans[i] = mask[i]
-		}
+		return parseChMask(0, mask[:]...), nil
 	case 6:
-		for i := uint8(0); i < 16; i++ {
-			chans[i] = true
-		}
-	default:
-		return nil, errUnsupportedChMaskCntl.WithAttributes("chmaskcntl", cntl)
+		return parseChMask(0,
+			true, true, true, true, true, true, true, true,
+			true, true, true, true, true, true, true, true,
+		), nil
 	}
-	return chans, nil
+	return nil, errUnsupportedChMaskCntl.WithAttributes("chmaskcntl", cntl)
 }
 
 func parseChMask72(mask [16]bool, cntl uint8) (map[uint8]bool, error) {
-	chans := make(map[uint8]bool, 72)
 	switch cntl {
-	case 0, 1, 2, 3, 4:
-		for i := uint8(0); i < 72; i++ {
-			chans[i] = (i >= cntl*16 && i < (cntl+1)*16) && mask[i%16]
-		}
+	case 0, 1, 2, 3:
+		return parseChMask(cntl*16, mask[:]...), nil
+	case 4:
+		return parseChMask(64, mask[0:8]...), nil
 	case 5:
-		for i := uint8(0); i < 64; i++ {
-			chans[i] = mask[i/8]
-		}
-		for i := uint8(64); i < 72; i++ {
-			chans[i] = mask[i-64]
-		}
-	case 6, 7:
-		for i := uint8(0); i < 64; i++ {
-			chans[i] = cntl == 6
-		}
-		for i := uint8(64); i < 72; i++ {
-			chans[i] = mask[i-64]
-		}
-	default:
-		return nil, errUnsupportedChMaskCntl.WithAttributes("chmaskcntl", cntl)
+		return parseChMask(0,
+			mask[0], mask[0], mask[0], mask[0], mask[0], mask[0], mask[0], mask[0],
```

Comment:
Abstract this away using `repeat`.

Reply:
No. This would require 7 chained appends, which would only obfuscate behavior.